### PR TITLE
feat: add workflow to publish code documentation to the Github Wiki upon package releases

### LINKED ISSUE #117: Generate Github/Markdown documentation.
Currently this repository is set up to use [Sphinx](https://www.sphinx-doc.org/en/master/) to build HTML documentation (or whichever other output format [Sphinx provides](https://www.sphinx-doc.org/en/master/usage/builders/index.html)). Considering this repo is hosted on Github, I think it follows naturally that we should be able to generate documentation in Markdown and publish the generated documentation here… 🤔

There are a few tools that generate Markdown from Python docstrings, with or without Sphinx ([older Stackoverflow question](https://stackoverflow.com/questions/13396856/markdown-output-for-sphinx-based-documentation)):
- [NiklasRosenstein/pydoc-markdown](https://github.com/NiklasRosenstein/pydoc-markdown)
- [ml-tooling/lazydocs](https://github.com/ml-tooling/lazydocs)
- [clayrisser/sphinx-markdown-builder](https://github.com/clayrisser/sphinx-markdown-builder)
- [mkdocs/mkdocs/](https://github.com/mkdocs/mkdocs/)

It would then make sense to integrate a “generate documentation” step into our [release Action](https://github.com/jenstroeger/python-package-template/blob/main/.github/workflows/check-and-release-main.yaml), and either find a way to publish the generated Markup directly into the repository or as a [Github Wiki](https://docs.github.com/en/communities/documenting-your-project-with-wikis/about-wikis) (which is its own repo), or use e.g. the [uibcdf/action-sphinx-docs-to-gh-pages](https://github.com/uibcdf/action-sphinx-docs-to-gh-pages) Action to publish to [Github Pages](https://pages.github.com/).

Lastly, perhaps it would also make sense for this repository to opine/dictate a docstring style (e.g. based on [Google’s Style Guide](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)) and check docstrings using Sphinx’s [Napoleon](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/) extension)?

### PR DESCRIPTION
**Pull request by jenstroeger**
This PR closes #117 and duplicates @lecram’s PR https://github.com/jenstroeger/python-package-template/pull/224 because I just couldn’t figure out how to push changes to his remote fork. The [Checkout a branch from a fork](https://github.com/orgs/community/discussions/23445) discussion was helpful but stopped short. So here we are…

This new PR contains all of @lecram’s original commits, rebased on the latest changes to `staging` and I added all of the PR feedback.

Things to do:

- [x] Update [README](https://github.com/jenstroeger/python-package-template/blob/main/README.md); and
- [x] With issue #355 and PR #398 in mind, make this a _reusable_ workflow as well (in particular, pass the artifacts).

### CODE REVIEW

**Comment by jenstroeger** [comment]
@behnazh this PR is ready for review.

@lecram once approved it would be great if you could squash-and-merge this PR because it’s mostly your work and I’d like you to take credit for it in this repo’s history.

File: .github/workflows/_wiki-documentation.yaml
Comment (behnazh) [inline .github/workflows/_wiki-documentation.yaml:44]: If the token is only required for private repositories, why not to comment it here and ask users to uncomment if they have a private repo

Comment (jenstroeger) [reply .github/workflows/_wiki-documentation.yaml:44]: Commit https://github.com/jenstroeger/python-package-template/pull/396/commits/0ee1aa67dd8572618b21836f8d5f0017b35740f4.

Comment (jenstroeger) [reply .github/workflows/_wiki-documentation.yaml:44]: That’s commenting/un-commenting in three places then…

https://github.com/jenstroeger/python-package-template/blob/798786b40760032dc0f218e8ac630a7f51739d7a/.github/workflows/release.yaml#L246-L247

https://github.com/jenstroeger/python-package-template/blob/798786b40760032dc0f218e8ac630a7f51739d7a/.github/workflows/_wiki-documentation.yaml#L24-L26

https://github.com/jenstroeger/python-package-template/blob/798786b40760032dc0f218e8ac630a7f51739d7a/.github/workflows/_wiki-documentation.yaml#L47-L48

Not sure it’s worthwhile, and maybe it’s just easier to always ask for the token?

**Review by behnazh — APPROVED** [review]

File: .github/workflows/_wiki-documentation.yaml
Comment (jenstroeger) [reply .github/workflows/_wiki-documentation.yaml:44]: And one more note: the `REPO_ACCESS_TOKEN` is also required by the Release workflow to push the bump commit, so we’re not asking anything new.

**Pull request merged**
